src/revlanguage/functions/phylogenetics/ratematrix/Func_GammaRateModel.cpp

```diff
@@ -0,0 +1,186 @@
+{
+}
+
+
+/**
+ * The clone function is a convenience function to create proper copies of inherited objected.
+ * E.g. a.clone() will create a clone of the correct type even if 'a' is of derived type 'b'.
+ *
+ * \return A new copy of the process.
+ */
+Func_GammaRateModel* Func_GammaRateModel::clone( void ) const
+{
+    return new Func_GammaRateModel( *this );
+}
+
+
+Core::TypedFunction< Core::MixtureModel >* Func_GammaRateModel::createFunction( void ) const
+{
+    Core::TypedDagNode< Core::MixtureModel >* submodel = dynamic_cast<const MixtureModel &>( this->args[0].getVariable()->getRevObject() ).getDagNode();
+    Core::TypedDagNode< double >* a = dynamic_cast<const RealPos &>( this->args[1].getVariable()->getRevObject() ).getDagNode();
+    Core::TypedDagNode< long >* nCats = dynamic_cast<const Integer &>( this->args[2].getVariable()->getRevObject() ).getDagNode();
+    Core::TypedDagNode< Core::Boolean >* median = dynamic_cast<const RlBoolean &>( this->args[3].getVariable()->getRevObject() ).getDagNode();
+
+    return Core::generic_function_ptr< Core::MixtureModel >( GammaRateModelFunc, submodel, a, nCats, median);
+}
+
+
+/* Get argument rules */
+const ArgumentRules& Func_GammaRateModel::getArgumentRules( void ) const
+{
+    static ArgumentRules argumentRules = ArgumentRules();
+    static bool          rules_set = false;
+
+    if ( !rules_set )
+    {
+        argumentRules.push_back( new ArgumentRule( "submodel"       , MixtureModel::getClassTypeSpec(), "Sub-model.", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
+        argumentRules.push_back( new ArgumentRule( "alpha"          , RealPos::getClassTypeSpec(), "The alpha parameter of the gamma distribution.", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
+        argumentRules.push_back( new ArgumentRule( "n"              , Integer::getClassTypeSpec(), "The number of bins to approximate the gamma distirbution.", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY, new Integer(4) ) );
+        argumentRules.push_back( new ArgumentRule( "median"         , RlBoolean::getClassTypeSpec(), "Should we use the median (or the mean)", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY, new RlBoolean(true) ) );
+
+        rules_set = true;
+    }
+
+    return argumentRules;
+}
+
+
+const std::string& Func_GammaRateModel::getClassType(void)
+{
+
+    static std::string rev_type = "Func_GammaRateModel";
+
+    return rev_type;
+}
+
+
+/* Get class type spec describing type of object */
+const TypeSpec& Func_GammaRateModel::getClassTypeSpec(void)
+{
+
+    static TypeSpec rev_type_spec = TypeSpec( getClassType(), new TypeSpec( Function::getClassTypeSpec() ) );
+
+    return rev_type_spec;
+}
+
+
+/**
+ * Get the primary Rev name for this function.
+ */
+std::string Func_GammaRateModel::getFunctionName( void ) const
+{
+    // create a name variable that is the same for all instance of this class
+    std::string f_name = "fnGamma";
```

Comment:
This definitely needs a different names. Better would be `fnGammaRates` or `fnGammaASRV`. For a function with name `fnGamme` I would expect either the Gamma function, or the Gamma distribution (e.g., pdf of Gamma).

Reply:
Based on the poll, I've changed `fnGamma` -> `fnGammaASRV`.